src/abi/typeFormulaParser.ts

```diff
@@ -83,6 +81,11 @@ export class TypeFormulaParser {
     private acquireTypeWithParameters(stack: any[]): TypeFormula {
         const typeParameters = this.acquireTypeParameters(stack);
         const typeName = stack.pop();
+
+        if (typeName === "ManagedDecimal" || typeName === "ManagedDecimalSigned") {
+            const typeFormula = new TypeFormula(typeName, [], typeParameters[0].name);
```

Comment:
Can define a separate variable above, such as `const metadata = typeParameters[0].name` - more obvious that we're treating the name of the first "type parameter" not as a type parameter, but as a special metadata (new concept).

Reply:
done :+1: